astropy/nddata/utils.py

```diff
@@ -194,3 +194,125 @@ def subpixel_indices(position, subsampling):
     # Get decimal points
     fractions = np.modf(np.asanyarray(position) + 0.5)[0]
     return np.floor(fractions * subsampling)
+
+
+def downsample(data, block_size, usemean=False):
+    """
+    Downsample (i.e. rebin) a 1D or 2D data array by block summation.
+
+    If the ``data`` size along any axis is not a whole multiple of
+    ``block_size``, then ``data`` will be trimmed and the extra
+    rows/columns will not be included in the output.
+
+    Parameters
+    ----------
+    data : array_like (1D or 2D)
+        The data to be resampled.
+
+    block_size : int
+        Integer downsampling factor along each axis.
+
+    usemean : bool, optional
+        If `True`, then the block-summed data will be divided by
+        ``block_size**N``, where ``N`` is the ``data`` dimension.
+        ``usemean=False`` (the default) will will conserve the data sum.
+
+    Returns
+    -------
+    output : array_like
+        The resampled data.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.nddata.utils import downsample
+    >>> data = np.arange(6)
+    >>> downsample(data, 2)
+    array([1, 5, 9])
+
+    >>> downsample(data, 2, usemean=True)
+    array([ 0.5,  2.5,  4.5])
+
+    >>> data = np.arange(16).reshape((4, 4))
+    >>> downsample(data, 2)
+    array([[10, 18],
+           [42, 50]])
+    """
+
+    data = np.asanyarray(data)
+    block_size = int(block_size)
+    size_new = np.array(data.shape) // block_size
+    size_init = size_new * block_size
+    if data.ndim == 1:
+        output = data[:size_init[0]].reshape(size_new[0],
+                                             block_size).sum(axis=1)
+    elif data.ndim == 2:
+        output = data[:size_init[0], :size_init[1]].reshape(
+            size_new[0], block_size, size_new[1], block_size).sum(axis=(1, 3))
```

Comment:
Instead of `sum ` this could be a parameter that the use could set to any jumpy ufunc.